### DISCUSSION TITLE
Updated Node Version on Forem

### DISCUSSION THREAD

**Discussion by cmgorton** [forem/forem#15925] [Announcements]
With this [PR](https://github.com/forem/forem/pull/15522) we have upgraded the **node.js** version on Forem from version 14.x to to 16.x (current LTS version). 

This change may cause an error if you have a branch of Forem as a work in progress. You will likely be notified that you have the wrong version of Node. You will need to update your node version with the following command:
`nvm install --lts`

As a further heads up we will also be upgrading the container base image from Fedora 34 to [Fedora 35](https://fedoramagazine.org/announcing-fedora-35/). If you're not using docker for development this should be a non-issue.

**Comment by cmgorton** [comment]
Error seen after running yarn or bin/setup:
``` bash
error dev.to@1.0.0: The engine "node" is incompatible with this module. Expected version "16.x". Got "14.18.1"
```

**Comment by djuber** [comment]
There is a note in the forem docs about issues installing canvas on an M1 mac (seen by some of our internal team today) - if you see node-pre-gyp errors installing, follow the brew install suggestion in the last solution on the installation page:

https://developers.forem.com/getting-started/installation/mac#possible-error-messages 

specifically, these libraries need to be installed to build canvas from source successfully:

```sh
brew install pkg-config cairo pango libpng jpeg giflib librsvg
```

**Comment by djuber** [comment]
There was a step in the bin/setup script to remove any (node 14) canvas compiled files, this [has been removed](https://github.com/forem/forem/pull/15966). I don't expect this will impact contributors, but if it does, removing the node_modules/canvas directory, and running yarn again, should bring in the new version of these files.
